### PR TITLE
Refactor: Use isinstance() for type checking to enhance flexibility i…

### PR DESCRIPTION
**Pull request by alphamarket**
**Refactor: Use `isinstance()` for type checking to enhance flexibility in workspace configuration**  

- Replaced direct class name checks with `isinstance()` to improve extensibility and maintainability.  
- Adjusted type handling for `WorkspaceSettings`, `DockerResources`, and `AwsResources` to ensure compatibility with future client-side customizations.  
- Streamlined object identification in `get_workspace_objects_from_file` and related functions for better readability and reduced redundancy.  
- Removed unnecessary imports and consolidated resource checks to align with updated type-checking logic.  

These changes make the workspace configuration process more robust and adaptable to evolving client requirements.

### CODE REVIEW

File: phi/workspace/config.py [hunk 1]
```diff
@@ -391,17 +401,17 @@ def get_resources_from_file(
         for obj_name, obj in workspace_objects.items():
             _obj_type = obj.__class__.__name__
```
Comment (dirkbrnd) [inline phi/workspace/config.py:402]: Can probably remove this line as well

File: phi/workspace/config.py [hunk 2]
```diff
@@ -289,20 +297,20 @@ def get_resources(
             for obj_name, obj in workspace_objects.items():
                 _obj_type = obj.__class__.__name__
```
Comment (dirkbrnd) [inline phi/workspace/config.py:298]: Remove this line

**Comment by alphamarket** [comment]
@dirkbrnd done.

**Review by dirkbrnd — APPROVED** [review]
Thanks for the addition!
LGTM

**Pull request merged**